adapter/oracle/adapter.go

```diff
@@ -104,6 +104,7 @@ func (a *Adapter) GetTable(tableName string) (*db.Table, error) {
 	}
 	table.Indexes = indexes
 
+	defer stmt.Close()
```

Comment:
If we write `defer` at the end of a `func`, this will not be executed if an error occurs earlier than that.

Therefore, please write it as early as possible.

e.g.
```go
	stmt, err := a.db.Preparex(a.db.Rebind(sql))
	if err != nil {
		return nil, err
	}
	defer stmt.Close() // <- Move to here!
```

Reply:
ok, I will add it after the stmt.Select and err check.

	stmt, err := a.db.Preparex(a.db.Rebind(sql))
	if err != nil {
		return nil, err
	  }
	  var rows []allIndColumns
	  err = stmt.Select(&rows, indexName)
	  if err != nil {
		return nil, err
	  }
	  defer stmt.Close()

Reply:
I see

Reply:
I can also move it after the stmt.Select so as to close the prepared statement in either cases

    stmt, err := a.db.Preparex(a.db.Rebind(sql))
    if err != nil {
	    return nil, err
      }
      var rows []allIndColumns
      
      err = stmt.Select(&rows, indexName)
      defer stmt.Close()

      if err != nil {
	    return nil, err
      }

Reply:
Updated as discussed